### PR TITLE
Implement a pairwise comparison of two segments

### PR DESCRIPTION
**Pull request by snukky**
In the comparison view, allows for a lightweight annotations of segment pairs with 'A<B', 'A>B' or 'A=B' label.

Changes proposed in the pull request:
- Added `PairwiseRanking` model
- Sending an annotation on a change of the select box using Ajax
- Not using JQuery Slim to have Ajax supported 

To do:
- Showing proper error messages
- Showing which segments already have annotations
- Exporting the annotations
- Optimizations

This PR is made on top of #25.

@AppraiseDev/core-team

### CODE REVIEW

**Comment by cfedermann** [comment]
@snukky: `ModuleNotFoundError: No module named 'pylint_django'` -- spurious issue or did you expect this to break?

File: evaluation/templates/comparison/submission.html [hunk 1]
```diff
@@ -30,8 +30,8 @@ <h3>Submission {{submission}}</h3>
             <thead>
               <tr>
                 <th>Line</th>
-                <th>Source</th>
-                <th>Translation</th>
+                <th>Source segments</th>
+                <th>Translations</th>
```
Comment (cfedermann) [inline evaluation/templates/comparison/submission.html:34]: @snukky: use `Target segments` instead? Or just `Source|Target`?

File: evaluation/views.py [hunk 2]
```diff
@@ -127,6 +131,9 @@ def submission(request, sub_id=None):
 def compare(request, sub_a_id=None, sub_b_id=None):
     """Renders vertical or horizontal comparison between two submissions."""
 
+    if request.method == "POST" and sub_a_id is None and sub_b_id is None:
```
Comment (cfedermann) [inline evaluation/views.py:134]: @snukky: what's the rationale? Don't you need the submission IDs for storing a comparison result?

File: evaluation/views.py [hunk 3]
```diff
@@ -182,15 +198,48 @@ def compare(request, sub_a_id=None, sub_b_id=None):
...
+def submit_rank(request):
+    """Handles Ajax request to create a PairwiseRanking object."""
+    # Accept POST requests only
+    if request.method != "POST":
+        return
+
+    msg = []
+
+    _, _, ocelot_team_token = _get_team_data(request)
+    sub_a_id = int(request.POST.get('submission_a_id'))
```
Comment (cfedermann) [inline evaluation/views.py:224]: @snukky: OK, you don't.

File: ocelot/urls.py [hunk 4]
```diff
@@ -52,6 +53,7 @@
     path('welcome', welcome, name='welcome-view'),
     # evaluation app
     path('compare/<sub_a_id>/<sub_b_id>', compare, name='compare-view'),
+    path('compare', compare, name='compare-view'),
```
Comment (cfedermann) [inline ocelot/urls.py:56]: @snukky: this really is the `submit-rank-for-comparison` view... The name is confusing.

**Review by cfedermann — CHANGES_REQUESTED** [review]
@snukky: OK, a last one for discussion today 👍

**Comment by snukky** [comment]
No, I don't plan to merge this as it might have already diverged too much from the main branch, but I would like to keep this pull request as a draft for having a reminder. Most of the code from this can be re-used once we're back into this enhancement.

**Pull request closed without merging**